src/hooks/useContextAnimation.ts

```diff
@@ -0,0 +1,41 @@
+import { useSelector } from 'react-redux'
+import Path from '../@types/Path'
+import isContextViewActive from '../selectors/isContextViewActive'
+import isDescendantPath from '../util/isDescendantPath'
+
+/** Return an animation name for a thought that is appearing or disappearing as a result of toggling a context view.
+ * This is used to display the animation in TreeNode, and also to determine whether thought annotations should delay
+ * their position calculation until after the animation has completed (#3352).
+ */
+const useContextAnimation = (path: Path, isAppearing?: boolean) => {
```

Comment:
This hook is too thin to justify the abstraction, as it is just a wrapper around `useSelector`. It would be better to factor out the selector and use `useSelector(contextAnimation(path, isAppearing))` as needed.